crates/services/p2p/src/p2p_service.rs

```diff
@@ -1241,7 +1288,12 @@ mod tests {
                     tracing::info!("Node A Event: {:?}", node_a_event);
                 },
                 node_b_event = node_b.next_event() => {
-                    if let Some(FuelP2PEvent::GossipsubMessage { topic_hash, message, .. }) = node_b_event.clone() {
+                    if let Some(FuelP2PEvent::GossipsubMessage { topic_hash, message, message_id, peer_id }) = node_b_event.clone() {
+                        // Message Validation must be reported
+                        // If it's `Accept`, Node B will propagate the message to Node C
+                        // If it's `Ignore` or `Reject`, Node C should not receive anything
+                        let msg_acceptance = to_message_acceptance(&acceptance);
+                        let _ = node_b.report_message_validation_result(&message_id, &peer_id, msg_acceptance);
```

Comment:
Hmm, I guess it is better to test it on the `service.rs` layer than on `p2p_service.rs`. In this case, you don't need to simulate the logic of the service and we will have a test to verify that service is gossiping data.

Reply:
I don't get the benefit, these are unit tests for `p2p_service` and we should have as many tests as possible for all kinds of scenarios where there multiple p2p services

Reply:
I meant we still need to test that the `service.rs` gossips transactions. Because `p2p_service.rs` is part of the `service.rs`, instead of unit testing it here, maybe it makes more sense to test it in the `service.rs`. In this case, you don't need to simulate the sending of `acceptance`.

Previously this test only checked the `publish_message` method. Now it checks `publish_message` and `report_message_validation_result`. Maybe we can test `report_message_validation_result` in a more simple setup?

Reply:
The test confirms 2 things:
- publish message works (A sends to B, B receives it)
- a) message propagation works on `Accept` report (B receives the message and sends it to C)
- b) message is not propagated to C if the report is not `Accept`

In order to test either a) or b) we need to `publish_message` anyway hence we test both things in the same test(s).
`report_message_validation_result` is useless without actual publishing.

`service.rs` should test it's own methods, channels etc, but p2p_service should be tested in isolation that it works as a unit regardless of its wrapper (`service.rs`).